packages/block-library/src/paragraph/editor.scss

```diff
@@ -11,12 +11,7 @@
 	}
 }
 
-.block-editor-block-list__block[data-type="core/paragraph"] p {
-	min-height: $empty-paragraph-height / 2;
-	line-height: $editor-line-height;
-}
-
 // Overwrite the inline style to make the height collapse when the paragraph editable gets focus.
 .block-editor-block-list__block[data-type="core/paragraph"] .has-drop-cap:focus {
-	min-height: $empty-paragraph-height / 2 !important;
+	min-height: auto !important;
```

Comment:
`auto` works just as well here, to autoresize the paragraph if the drop cap disappears and the inline style needs to be overridden.